docs/testing/e2e-testing-standards.md

```diff
@@ -0,0 +1,309 @@
+In ownCloud, we use Playwright for webUI test automation. We benefit from lower barriers to entry, readability, and usability when test standards are consistent across repositories. For example:
+
+- **Reusability:** Enhances reusability by making it easier to reuse functions, locators, shared steps, and other test code. Finding the necessary components and functionalities will be simple for someone working on a test resulting in reduced duplication and the requirement for rework after the initiation of a code review.
+
+- **Facilitates reviews:** Code can be examined more quickly. Since you can quickly determine what the test code is doing, it eases the mental strain of code review.
+
+- **Faster onboarding:** By following naming conventions, new contributors are onboarded more rapidly and feel comfortable enough to contribute to the codebase.
+
+Here are the test standards and guidelines we adhere to when creating Playwright tests at ownCloud.
+
+## Folder Structure:
+- `tests/:`
+
+    - `e2e/`: Main folder containing all (end-to-end) E2E test-related files.
+    
+      - `cucumber/`: Main folder containing all Cucumber(BDD) test-related files.
+
+        - `features/`: Contains Gherkin feature files.
+
+            - `<test-suite-folder>/`: Collection house for "**related"** feature files.
+
+                - `<aFeatureFile>.feature`: A feature file.
+
+        - `steps/`: Holds the step definition files for mapping Gherkin steps to code.
+
+            - `<stepDefinition>.ts`: Step definitions for each feature.
+
+      - `support/`: Playwright (Test implementation)
+
+        - `api/`: Contains API-related test files and configurations.
+          
+          - `<api-folder>/ `: Specific API tests for a particular service.
+
+        - `objects/`: Contains the Page Object classes.
+
+            - `<specific-page-object-folder>/`: Collection house for related page objects for each webpage or component.
+
+                - `<individualPageObject>.ts`: Page Object for each webpage or component.
+
+        - `utils/`: Utility functions and common helpers.
+
+            - `helpers.ts`: Common utility functions (e.g., date formatting, data generation).
+
+        - `config/`: Configuration files for Playwright and other tools.
+
+            - `playwright.config.ts`: Playwright configuration.
+
+        - `hooks/`: Cucumber hooks for setting up and tearing down test environments.
+
+            - `hooks.ts`: Contains `Before`, `After`, and other lifecycle hooks.
+
+        - `reports/`: Generated test reports (e.g., HTML, JSON).
+
+            - `screenshots/`: Captured screenshots during test execution.
+
+            - `videos/`: Recorded videos of test runs.
```

Comment:
this should be directly inside `e2e/` directory

Reply:
OK and what about `test-data/filesForUpload`?

Reply:
OK and what about `test-data/filesForUpload`?

Reply:
that should be fine there